azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/http/http_transport.py

```diff
@@ -0,0 +1,149 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import logging
+import requests
+from datetime import date
+from azure.iot.hub.devicesdk.transport.abstract_transport import AbstractTransport
+from azure.iot.hub.devicesdk.message import Message
+
+"""
+The below import is for generating the state machine graph.
+"""
+# from transitions.extensions import LockedGraphMachine as Machine
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPTransport(AbstractTransport):
+    def __init__(self, auth_provider):
+        """
+        Constructor for instantiating a transport
+        :param auth_provider: The authentication provider
+        """
+        AbstractTransport.__init__(self, auth_provider)
```

Comment:
use super()

Reply:
I don't think it would be possible as AbstractTransport would need to be modified to have object as base class

---
In reply to: [250814837](https://github.com/Azure/azure-iot-sdk-python-preview/pull/41#discussion_r250814837) [](ancestors = 250814837)